hazelcast/src/main/java/com/hazelcast/query/impl/TypeConverters.java

```diff
@@ -91,15 +91,20 @@ Comparable convertInternal(Comparable value) {
     static class SqlTimestampConverter extends BaseTypeConverter {
         @Override
         Comparable convertInternal(Comparable value) {
-            if (value instanceof Timestamp) {
-                return value;
+            if (value instanceof java.sql.Timestamp) {
+                return ((java.sql.Timestamp) value).getTime();
```

Comment:
Seems `((java.sql.Timestamp) value).getTime()` is not enough for a true comparison. Please see below method from Timestamp source, it also make use of `nanos` in case of equality. 

```
public int compareTo(Timestamp ts) {
        long thisTime = this.getTime();
        long anotherTime = ts.getTime();
        int i = (thisTime<anotherTime ? -1 :(thisTime==anotherTime?0 :1));
        if (i == 0) {
            if (nanos > ts.nanos) {
                    return 1;
            } else if (nanos < ts.nanos) {
                return -1;
            }
        }
        return i;
    }
```